### ISSUE TITLE
Unexpected base64 parsing behavior

### ISSUE THREAD

**Issue by Ratniex** [MikeMcl/bignumber.js#198]
When setting the alphabet to base64 characters parsing strings that start with a space and then a plus sign (e.g. ' +a') then the plus is not interpreted as a character from the alphabet, but as an indicator of the number being positive.

```
const BigNumber = require('bignumber.js')

BigNumber.config({ALPHABET: '0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ+/'})

s = ' +a'
console.log(BigNumber(s, 64).toString()) // prints 10
s2 = '+a'
console.log(BigNumber(s, 64).toString()) // prints 3978
```

**Comment by MikeMcl** [comment]
I can't reproduce that behaviour.

Open the console at the [documentation page](http://mikemcl.github.io/bignumber.js/):

```js
BigNumber.config({ALPHABET: '0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ+/'})

s = ' +a'
console.log(BigNumber(s, 64).toString()) // prints 10
s2 = '+a'
console.log(BigNumber(s, 64).toString()) // prints 10
```

Which version are you using?

**Comment by Ratniex** [comment]
you should also consider trimming the parameter passed to constructor if it is a string and disallow whitespace characters in the alphabet.

**Comment by MikeMcl** [comment]
Fixed in *v8.0.0*.
Thank you for your assistance in this matter.
